src/ResourceManagement/AppService/Domain/IFunctionApp.cs

```diff
@@ -95,5 +96,21 @@ public interface IFunctionApp  :
         /// <param name="functionName">The name of the function.</param>
         /// <return>The function key.</return>
         Task<System.Collections.Generic.IReadOnlyDictionary<string,string>> ListFunctionKeysAsync(string functionName, CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary>
+        /// List the functions
+        /// </summary>
+        IEnumerable<IFunctionEnvelope> ListFunctions();
```

Comment:
Should be IReadonlyList?

Reply:
IEnumerable is already readonly for the underlying collection.

I am referring to this interface on the method signature of returned resource list (my feeling is that it more like an operation on object, than a property of object):

`    public interface ISupportsListing<T>
    {
        /// <summary>
        /// Lists all the resources of the specified type in the currently selected subscription.
        /// </summary>
        /// <returns>list of resources</returns>
        IEnumerable<T> List();

        /// <summary>
        /// Lists all the resources of the specified type in the currently selected subscription.
        /// </summary>
        /// <returns>list of resources</returns>
        Task<IPagedCollection<T>> ListAsync(bool loadAllPages = true, CancellationToken cancellationToken = default(CancellationToken));
    }
`

I will change to IReadOnlyList if you feel it be more consistent with the codebase.

Reply:
For API, it is important to provide consistent interface/API exposed to the customer.